.github/workflows/automate_javadoc.yml

```diff
@@ -1,27 +1,29 @@
-# SPDX-FileCopyrightText: 2022 RTE FRANCE
+# SPDX-FileCopyrightText: 2022 2023 RTE FRANCE
 #
 # SPDX-License-Identifier: Apache-2.0
 name: Continuous Deployment - JavaDocs
 
 on:
-  push:
-    branches:
-      - 'feat/javadoc'
+  pull_request_target:
+    types: [ labeled, closed ]
+
 jobs:
   build:
+    if: ${{  github.event.pull_request.base.ref == 'develop' && github.actor != 'dependabot[bot]' && github.event.pull_request.merged && contains(github.event.pull_request.labels.*.name, 'javadoc') }}
     name: Java Docs
     runs-on: ubuntu-latest
     env:
-      from_branch: feat/javadoc
-      target_branch: feat/117_Documentation_SCD_Generation_Process
+      DEPLOYMENT_BRANCH: gh-pages
```

Comment:
To be discussed, we could try to ovoid many branches to manage if possible

Reply:
Yes, there is only the `gh-pages` branch here. 

![Capture d’écran du 2023-02-02 15-59-35](https://user-images.githubusercontent.com/76203423/216359952-0c0a0fb4-fb57-4f58-b471-f1522437173e.png)

See (https://github.com/com-pas/compas-sct/issues/159#issuecomment-1406709340) for more detail.

Reply:
ok